Tests/NIOCoreTests/AsyncSequences/NIOAsyncSequenceTests.swift

```diff
@@ -391,8 +391,14 @@ final class NIOAsyncSequenceProducerTests: XCTestCase {
         let sequence = try XCTUnwrap(self.sequence)
         let task: Task<Int?, Never> = Task {
             let iterator = sequence.makeAsyncIterator()
-            return await iterator.next()
+            let value = await iterator.next()
+
+            // Sleeping here a bit to make sure we hit the 
+            try? await Task.sleep(nanoseconds: 500_000)
```

Comment:
Is 500 microseconds sufficient?

Reply:
Changed to 1 ms and 2 ms. That should be enough